rinse/core.py

```diff
@@ -231,5 +303,162 @@ def __init__(self):
 
 class WindowsInstallR(BaseInstallR):
 
-    def __init__(self):
-        raise NotImplementedError("Installation of R with rinse on Windows is not supported at this time.")
+    def __init__(self, version, method, name, path, repos, glbl, config_clear, config_keep, init, verbose):
+        super().__init__(path=path, version=version, repos=repos, method=method, name=name, init=init, verbose=verbose)
+        self.config_clear = config_clear
+        self.config_keep = config_keep
+        self.src_file_path = self.src_path
+        if glbl is not None:
+            self.global_interpreter(version=glbl)
+
+    def _url_download(self, url, filepath, filename):  
+        with open(filepath, "wb") as f:
+            try:
+                self.logger.info("Downloading %s" % filename)
+                response = requests.get(url, stream=True)
+                total_length = response.headers.get('content-length')
+
+                if not total_length:  # no content length header
+                    f.write(response.content)
+                else:
+                    dl = 0
+                    total_length = int(total_length)
+                    for data in response.iter_content(chunk_size=4096):
+                        dl += len(data)
+                        f.write(data)
+                        done = int(100 * dl / total_length)
+                        half_done = int(done/2)
+                        sys.stdout.write("\r[%s%s] %d%% complete" % ('=' * half_done, ' ' * (50-half_done), done))
+                        sys.stdout.flush()
+
+            except KeyboardInterrupt:
+                f.close()
+                self.logger.error("\n Ctrl-c pressed. Aborting!")
+                delete_cmd = "del %s" % filepath
+                system_cmd(cmd=delete_cmd, stdout=sp.PIPE, stderr=sp.STDOUT, shell=True)
+                sys.exit(0)
+
+        sys.stdout.write('\n')
+        self.logger.info("\n%s Downloaded Successfully" % filename)
```

Comment:
Just a thought here:

It may be smoother if the try/except happens outside of the file writing `with` statement.  Also, you can log as an exception to exit and use `remove` which has been imported from `os` instead of a system call.

```python
                f.close()
                remove(filepath)
                self.logger.exception("\n Ctrl-c pressed. Aborting!")
```

Reply:
OK I will look into it.